### ISSUE TITLE
GC Yough Times Differ & Inconsistant Question

### ISSUE THREAD

**Issue by Aiidoneus** [cpburnz/minecraft-prometheus-exporter#20]
Hello,
We have been using this for a while for our own graphs on our network and wanted to know why the mod Spark is reporting different results than what prometheus might be showing? Any ideas why this might be? 

From our graphs
![](https://i.imgur.com/NrT8FEP.png)
From Spark
![](https://i.imgur.com/Q9SHe6y.png)

**Comment by cpburnz** [comment]
The Prometheus library I'm using ([simpleclient_hotspot](https://github.com/prometheus/client_java/tree/parent-0.16.0/simpleclient_hotspot)) ultimately gets it's GC stats from Java's [GarbageCollectorMXBean](https://docs.oracle.com/en/java/javase/17/docs/api/java.management/java/lang/management/GarbageCollectorMXBean.html) via [ManagementFactory](https://docs.oracle.com/en/java/javase/17/docs/api/java.management/java/lang/management/ManagementFactory.html).

I created a test server, and it had 5 old generation GCs by the time I connected. Then that number only increased by 2 about 15 minutes later. It appears to be fairly rare. 

```
jvm_gc_collection_seconds_count{gc="G1 Young Generation",} 402.0
jvm_gc_collection_seconds_sum{gc="G1 Young Generation",} 2.336
jvm_gc_collection_seconds_count{gc="G1 Old Generation",} 7.0
jvm_gc_collection_seconds_sum{gc="G1 Old Generation",} 2.639
```

It looks like you're using the Minecraft Server Stats dashboard. The "GC time" graph uses the query:

```
increase(jvm_gc_collection_seconds_sum{application="$application", instance="$instance"}[$__interval])
```

So it only reports the number of GC events over each interval (60s on my instance), and your window looks to be about 20m. Your Spark screenshot is probably reporting the average of all events, not just over 1 minute intervals. It also says the average frequency of an old generation is about 1 hour, so you might see a single blip for the old generation if you increase your window to 3 hours. For the young generation my guess is periods of no connected players is biasing the average to be too low compared to high activity. I get 1 young generation every few seconds in the nether with no other mods (except for the Prometheus Exporter).
